compiler/next/include/chpl/queries/ID.h

```diff
@@ -80,34 +90,33 @@ class ID final {
 
   /**
     returns 'true' if the AST node with this ID contains the AST
-    node with the other ID, including if they refer to the same AST node. 
+    node with the other ID, including if they refer to the same AST node.
    */
-  bool contains(const ID other) const;
+  bool contains(const ID& other) const;
 
   /**
     compare this ID with another ID
       result  < 0 if this < other
       result == 0 if this == other
       result  > 0 if this > other
    */
-  int compare(const ID other) const;
+  int compare(const ID& other) const;
 
-  inline bool operator==(const ID other) const {
-    return this->symbolPath_ == other.symbolPath_ &&
-           this->postOrderId_ == other.postOrderId_;
+  bool operator==(const ID& other) const {
+    return symbolPath_ == other.symbolPath_ &&
+          postOrderId_ == other.postOrderId_;
   }
-  inline bool operator!=(const ID other) const {
-    return this->symbolPath_ != other.symbolPath_ ||
-           this->postOrderId_ != other.postOrderId_;
+
+  bool operator!=(const ID& other) const {
+    return !(*this == other);
   }
 
   bool isEmpty() const {
     return symbolPath_.isEmpty();
   }
 
   size_t hash() const {
-    std::hash<int> hasher;
-    return hash_combine(symbolPath_.hash(), hasher(postOrderId_));
+    return hash_combine(symbolPath_.hash(), std::hash<int>{}(postOrderId_));
```

Comment:
Please put it back. The earlier version was not much longer but doesn't require level 10 C++ skills.

Reply:
ok